components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java

```diff
@@ -280,26 +282,43 @@ public Response callback(@Context Request request, @PathParam("appName") String
                 String part2 = accessToken.substring(accessToken.length() / 2);
                 NewCookie cookieWithAppContext = AuthUtil
                         .cookieBuilder(AuthenticatorConstants.ACCESS_TOKEN_1, part1, appContext,
-                                true, false, "future");
+                                true, false, "");
+                authResponseBean.setPartialToken(part1);
                 NewCookie httpOnlyCookieWithAppContext = AuthUtil
                         .cookieBuilder(AuthenticatorConstants.ACCESS_TOKEN_2, part2, appContext,
-                                true, true, "future");
+                                true, true, "");
                 NewCookie restAPIContextCookie = AuthUtil
                         .cookieBuilder(APIConstants.AccessTokenConstants.AM_TOKEN_MSF4J, part2, restAPIContext,
-                                true, true, "future");
+                                true, true, "");
                 String authUser = authResponseBean.getAuthUser();
                 NewCookie authUserCookie = AuthUtil
                         .cookieBuilder(AuthenticatorConstants.AUTH_USER, authUser, appContext, true, false, "");
                 if (log.isDebugEnabled()) {
                     log.debug("Set cookies for " + appName + " application.");
                 }
                 // Redirect to the store/apis page (redirect URL)
-                URI targetURIForRedirection = new URI(storeConfigs.getApimBaseUrl() + appName);
-                return Response.status(Response.Status.FOUND)
-                        .header(HttpHeaders.LOCATION, targetURIForRedirection).entity(authResponseBean)
-                        .cookie(cookieWithAppContext, httpOnlyCookieWithAppContext,
-                                restAPIContextCookie, authUserCookie)
-                        .build();
+                URI targetURIForRedirection = new URI(appConfigs.getApimBaseUrl() + appName);
+                if (AuthenticatorConstants.PUBLISHER_APPLICATION.equals(appName)) {
+                    String authResponseBeanData = authResponseBean.getAuthUser() + "&id_token="
+                            + authResponseBean.getIdToken() + "&partial_token=" + authResponseBean.getPartialToken()
+                            + "&scopes=" + authResponseBean.getScopes() + "&validity_period="
+                            + authResponseBean.getValidityPeriod();
+                    URI redirectURI = new URI(appConfigs.getApimBaseUrl() + "publisher/login?user_name="
+                            + URLEncoder.encode(authResponseBeanData, "UTF-8")
+                            .replaceAll("\\+", "%20").replaceAll("%26", "&")
+                            .replaceAll("%3D", "="));
```

Comment:
We have to handle some of the characters separately as java.net.URLEncoder does not support them.